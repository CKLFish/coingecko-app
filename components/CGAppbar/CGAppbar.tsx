// import {useNavigationState} from '@react-navigation/native';
// import _ from 'lodash';
import React from 'react';
import {Appbar} from 'react-native-paper';

const CGAppbar: React.FC = () => {
  //   const navigationState = useNavigationState(state => state);

  return (
    <Appbar>
      {/* {hasPreviousRoute ? (
        <Appbar.BackAction
          onPress={() => {
            navigation.goBack();
          }}
        />
      ) : null} */}
      <Appbar.Content title={'Crypto'} />
    </Appbar>
  );
};

export default CGAppbar;
