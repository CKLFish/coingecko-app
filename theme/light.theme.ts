import {DefaultTheme} from 'react-native-paper';

const theme = {
  ...DefaultTheme,
  colors: {
    ...DefaultTheme.colors,
    // # For Up / Down / Unchanged
    up: 'green',
    down: 'red',
    unchange: 'grey',
    favourite: 'red',
  },
};

export default theme;
