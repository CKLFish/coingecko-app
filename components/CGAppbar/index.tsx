import CGAppbar from './CGAppbar';

export default CGAppbar;
