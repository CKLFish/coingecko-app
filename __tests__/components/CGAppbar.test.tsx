/**
 * @format
 */

import 'react-native';

import React from 'react';
import {Provider} from 'react-native-paper';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import CGAppbar from '../../components/CGAppbar';
import theme from '../../theme/light.theme';

it('renders CGAppbar', () => {
  const tree = renderer.create(
    <Provider theme={theme}>
      <CGAppbar />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
