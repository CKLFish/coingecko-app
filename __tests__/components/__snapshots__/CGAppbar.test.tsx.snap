// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CGAppbar 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6200ee",
        "elevation": 4,
        "flexDirection": "row",
        "height": 56,
        "paddingHorizontal": 4,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 4,
      }
    }
  >
    <View
      style={
        Object {
          "width": 48,
        }
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "paddingHorizontal": 12,
          },
          Array [
            false,
            Object {
              "alignItems": "center",
            },
            undefined,
          ],
        ]
      }
    >
      <Text
        accessibilityRole="header"
        accessibilityTraits="header"
        accessible={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "fontSize": 17,
              },
              undefined,
            ],
          ]
        }
      >
        Crypto
      </Text>
    </View>
    <View
      style={
        Object {
          "width": 48,
        }
      }
    />
  </View>
</View>
`;
